views/pages/AdminPage.jsx

```diff
@@ -1,696 +1,201 @@
-  }
-  // eslint-disable-next-line react/no-deprecated
-  async componentWillMount() {
+  const handleResetPassword = async (rowIndex) => {
     try {
-      const acl = await fetchSubjects()
-      this.setState(getCounts({ acl }))
-      const users = await api.users.loadAll()
-      this.setState({
-        users,
-        autocomplete: this.autocomplete({ users }),
-      })
-      const studies = await fetchStudiesAdmin()
-      this.setState({
-        studies: studies.map((suggestion) => ({
-          value: suggestion,
-          label: suggestion,
-        })),
-      })
-    } catch (err) {
-      console.error(err.message)
+      const resetKey = crypto.randomUUID()
+      const userId = getUserValues(rowIndex).uid
+      const userAttributes = { force_reset_pw: true, reset_key: resetKey }
+      const updatedUser = await api.admin.users.update(userId, userAttributes)
+
+      updateUserInForm(rowIndex, updatedUser)
+      setResetKeyValue(resetKey)
+      setOpenResetKeyModal(true)
+      handleNotification('Reset Key has been applied successfully')
+    } catch (error) {
+      handleNotification(error.message)
     }
   }
-  componentWillUnmount() {
-    window.removeEventListener('resize', this.handleResize)
-  }
-  handleDrawerToggle = () => {
-    this.setState((state) => ({ mobileOpen: !state.mobileOpen }))
+  const handleUserRole = (rowIndex) => {
+    setOpenAccessModal(true)
+    setCurrentRowIndex(rowIndex)
   }
-  autocomplete = ({ users }) =>
-    users.map((option) => ({
-      value: option.uid,
-      label:
-        /\S/.test(option.display_name) && option.display_name != undefined
-          ? option.display_name
-          : option.uid,
-    }))
-  handleTab = (value) => {
-    this.setState({
-      tab: value,
-    })
+  const handleClose = () => {
+    setResetKeyValue('')
+    setOpenResetKeyModal(false)
+    setOpenAccessModal(false)
+    setCurrentRowIndex(null)
   }
-  handleResize = () => {
-    this.setState({
-      width: window.innerWidth - this.state.marginWidth,
-      height: window.innerHeight - this.state.marginHeight,
-    })
+  const handleUpdateUserAccess = async (rowIndex) => {
+    try {
+      const userValues = getUserValues(rowIndex)
+      const userId = userValues.uid
+      const userAttributes = UsersModel.createUserFromFormValues(userValues)
+      const updatedUser = await api.admin.users.update(userId, userAttributes)
+      const userToFormValues = UsersModel.createFormValuesFromUser(updatedUser)
+
+      updateUserInForm(rowIndex, userToFormValues)
+      handleClose()
+      handleNotification('User updated successfully')
+    } catch (error) {
+      handleNotification(error.message)
+    }
   }
-  sort = ({ sortBy, sortDirection }) => {
-    const sortedList = this.sortList({ sortBy, sortDirection })
-    this.setState({
-      sortBy: sortBy,
-      sortDirection: sortDirection,
-      users: sortedList,
-    })
+  const handleUserUpdate = async (data, userId) => {
```

Comment:
this is very similar to the `handleUpdateUserAccess` method. Can we combine them?